### ISSUE TITLE
HTTP 500 Error when logging in.

### ISSUE THREAD

**Issue by thecmdradama** [orangehrm/orangehrm#1305]
**Environment details**
OrangeHRM version: 5.0
OrangeHRM source: Release build from github
Ubuntu: 21.10
PHP: 8.1.6
Database: MySQL  Ver 8.0.29-0ubuntu0.21.10.2
Web server: nginx version: nginx/1.18.0 (Ubuntu)

If applicable:
Browser: Chrome and Edge

**Describe the bug**
User is presented with a HTTP ERROR 500 page when logging in. URL is https://hrm.mydomain.com/web/index.php/auth/validate
Navigating back to the log in page and refreshing the page logs in correctly

**Expected behavior**
User navigates to https://hrm.mydomain.com/web/index.php/auth/login and log in as normal.
User is then taken to https://hrm.mydomain.com/web/index.php/pim/viewEmployeeList as normal

**What do you see instead:**
As mentioned above. When logging in, the user is presented with a HTTP 500 instead of logging as expected

In src/log/orangehrm.log when error occurs

```
[2022-05-30T10:43:47.242599+00:00] orangehrm.ERROR: An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'login_time' cannot be null [] []
[2022-05-30T10:43:47.242814+00:00] orangehrm.ERROR: #0 /var/www/html/orangehrm/src/vendor/doctrine/dbal/src/Connection.php(1768): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert() #1 /var/www/html/orangehrm/src/vendor/doctrine/dbal/src/Connection.php(1707): Doctrine\DBAL\Connection->handleDriverException() #2 /var/www/html/orangehrm/src/vendor/doctrine/dbal/src/Statement.php(184): Doctrine\DBAL\Connection->convertExceptionDuringQuery() #3 /var/www/html/orangehrm/src/vendor/doctrine/dbal/src/Statement.php(221): Doctrine\DBAL\Statement->execute() #4 /var/www/html/orangehrm/src/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php(276): Doctrine\DBAL\Statement->executeStatement() #5 /var/www/html/orangehrm/src/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(1120): Doctrine\ORM\Persisters\Entity\BasicEntityPersister->executeInserts() #6 /var/www/html/orangehrm/src/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(415): Doctrine\ORM\UnitOfWork->executeInserts() #7 /var/www/html/orangehrm/src/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(388): Doctrine\ORM\UnitOfWork->commit() #8 /var/www/html/orangehrm/src/plugins/orangehrmCorePlugin/Traits/ORM/EntityManagerHelperTrait.php(51): Doctrine\ORM\EntityManager->flush() #9 /var/www/html/orangehrm/src/plugins/orangehrmAuthenticationPlugin/Dao/LoginLogDao.php(33): OrangeHRM\Core\Dao\BaseDao->persist() #10 /var/www/html/orangehrm/src/plugins/orangehrmAuthenticationPlugin/Service/LoginService.php(63): OrangeHRM\Authentication\Dao\LoginLogDao->saveLoginLog() #11 /var/www/html/orangehrm/src/plugins/orangehrmAuthenticationPlugin/Controller/ValidateController.php(118): OrangeHRM\Authentication\Service\LoginService->addLogin() #12 /var/www/html/orangehrm/src/vendor/symfony/http-kernel/HttpKernel.php(152): OrangeHRM\Authentication\Controller\ValidateController->handle() #13 /var/www/html/orangehrm/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #14 /var/www/html/orangehrm/src/lib/framework/Framework.php(160): Symfony\Component\HttpKernel\HttpKernel->handle() #15 /var/www/html/orangehrm/web/index.php(42): OrangeHRM\Framework\Framework->handleRequest() #16 {main} [] []
[2022-05-30T10:43:48.319674+00:00] orangehrm.ERROR: Client error: `POST https://ospenguin.orangehrm.com` resulted in a `400 Bad Request` response [] []
[2022-05-30T10:43:48.319760+00:00] orangehrm.ERROR: #0 /var/www/html/orangehrm/src/vendor/guzzlehttp/guzzle/src/Middleware.php(69): GuzzleHttp\Exception\RequestException::create() #1 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}() #2 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler() #3 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}() #4 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Promise\TaskQueue->run() #5 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn() #6 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending() #7 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList() #8 /var/www/html/orangehrm/src/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending() #9 /var/www/html/orangehrm/src/vendor/guzzlehttp/guzzle/src/Client.php(187): GuzzleHttp\Promise\Promise->wait() #10 /var/www/html/orangehrm/src/vendor/guzzlehttp/guzzle/src/ClientTrait.php(95): GuzzleHttp\Client->request() #11 /var/www/html/orangehrm/src/plugins/orangehrmCorePlugin/Registration/Service/RegistrationAPIClientService.php(56): GuzzleHttp\Client->post() #12 /var/www/html/orangehrm/src/plugins/orangehrmCorePlugin/Registration/Processor/AbstractRegistrationEventProcessor.php(209): OrangeHRM\Core\Registration\Service\RegistrationAPIClientService->publishData() #13 /var/www/html/orangehrm/src/plugins/orangehrmCorePlugin/Registration/Subscriber/RegistrationEventPublishSubscriber.php(55): OrangeHRM\Core\Registration\Processor\AbstractRegistrationEventProcessor->publishRegistrationEvents() #14 /var/www/html/orangehrm/src/vendor/symfony/event-dispatcher/EventDispatcher.php(230): OrangeHRM\Core\Registration\Subscriber\RegistrationEventPublishSubscriber->onTerminateEvent() #15 /var/www/html/orangehrm/src/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\Component\EventDispatcher\EventDispatcher->callListeners() #16 /var/www/html/orangehrm/src/vendor/symfony/http-kernel/HttpKernel.php(94): Symfony\Component\EventDispatcher\EventDispatcher->dispatch() #17 /var/www/html/orangehrm/web/index.php(48): Symfony\Component\HttpKernel\HttpKernel->terminate() #18 {main} [] []
```

Any help would be appreciated.

**Comment by RajithaKumara** [comment]
Hi @thecmdradama,
Thanks for evaluating OrangeHRM 5.0 :relaxed:.
For the moment we did not fully test on MySQL 8.0. How you installer on MySQL 8.0?
Anyway, this seems a bug in the application. Do you need an immediate patch?

**Comment by RajithaKumara** [comment]
Please try below steps;

- Find `src/plugins/orangehrmAuthenticationPlugin/entity/LoginLog.php`
- Extract this [LoginLog.zip](https://github.com/orangehrm/orangehrm/files/8892325/LoginLog.zip) and replace `src/plugins/orangehrmAuthenticationPlugin/entity/LoginLog.php` file. 
- Make sure to backup the current code and database.

**Comment by RajithaKumara** [comment]
Thanks for reporting this issue.
Fixed with [OrangeHRM 5.1 release](https://sourceforge.net/projects/orangehrm/files/stable/5.1/orangehrm-5.1.zip/download)
